Sources/Account/Balance.swift

```diff
@@ -5,95 +5,146 @@
-    /// - `mobInt`: XXXXXXXXX (denominated in MOB)
-    /// - `picoFrac`: YYYYYYYYYYYY (denominated in picoMOB)
+    /// - `int`: XXXXXXXXX (denominated in MOB)
+    /// - `frac`: YYYYYYYYYYYY (denominated in picoMOB)
     ///
     /// It is necessary to break apart the values into 2 parts because the total max possible
     /// balance is too large to fit in a single `UInt64`, when denominated in picoMOB, assuming 250
     /// million MOB in circulation and assuming a base unit of 1 picoMOB as the smallest indivisible
     /// unit of MOB.
-    public var amountMobParts: (mobInt: UInt32, picoFrac: UInt64) {
+    public var amountParts: (int: UInt64, frac: UInt64) {
+        //
+        // >> Example math with significantDigits == 12 (MOB)
+        //
         // amount (picoMOB) = amountLow + amountHigh * 2^64
         //
+        // >> Now expand amountLow & amountHigh to "decimal" numbers
+        //
         // amountLowMobDec = amountLow / 10^12
         // amountHighMobDec = amountHigh * 2^64 / 10^12
         //
+        // >> where 10^12 is 10^(significantDigits)
+        //
         // amountMobDec = amountLowMobDec + amountHighMobDec
         //
+        // >> Now expand amountLowDec & amountHighDec to their Integer & Fractional parts
+        //
+        // >> amountLowDec = amountLowMobInt.amountLowPicoFrac
+        //
         // amountLowMobInt = floor(amountLow / 10^12)
         // amountLowPicoFrac = amountLow % 10^12
-
+        //
+        // >> amountHighDec = amountHighMobInt.amountHighPicoFrac
+        //
         // amountHighMobInt = floor((amountHigh * 2^64) / 10^12)
+        //
+        //                  >> factor out common 2^12 for now
+        //                  = floor((amountHigh * (2^52 * 2^12)) / (5^12 * 2^12)
+        //
+        //                  >> bitshift by 52 (same as multiply by 2^52)
+        //                  >> ... and bitshift number == (64 - significantDigits)
         //                  = floor((amountHigh << 52) / 5^12)
+        //
         // amountHighPicoFrac = (amountHigh * 2^64) % 10^12
-        //                   = ((amountHigh << 52) % 5^12) << 12
+        //
+        //                  >> re-apply the 2^12
+        //                  = ((amountHigh << 52) % 5^12) << 12
         //
         // amountPicoFracCarry = floor((amountLowPicoFrac + amountHighPicoFrac) / 10^12)
         //
         // amountMobInt = amountLowMobInt + amountHighMobInt + amountPicoFracCarry
         // amountPicoFrac = (amountLowPicoFrac + amountHighPicoFrac) % 10^12
 
-        let (amountLowMobInt, amountLowPicoFrac) = { () -> (UInt32, UInt64) in
-            // 10^12 = 1_000_000_000_000
-            let mobParts = amountPicoMobLow.quotientAndRemainder(dividingBy: 1_000_000_000_000)
-            return (UInt32(mobParts.quotient), mobParts.remainder)
+        
+        let significantDigits = tokenId.significantDigits
+        
+        let divideBy = UInt64(pow(Double(10), Double(significantDigits)))
```

Comment:
I feel like we should be using a precomputed value instead of using pow for every balance... since the divideBy could be calculated for every tokenId once instead of every time a Balance is created

Reply:
I think this is a premature optimization, if there is evidence that these calculation are causing noticeable slowdowns, id be willing to change my mind.

Reply:
we can revisit when we get to > 100 available tokens